src/data/nat/cast.lean

```diff
@@ -321,3 +321,18 @@ begin
 end
 
 end with_top
+
+namespace pi
+
+variables {α β : Type*}
+
+lemma nat_apply [has_zero β] [has_one β] [has_add β] :
```

Comment:
```suggestion
@[simp]
lemma nat_apply [has_zero β] [has_one β] [has_add β] :
```

Reply:
I think the `coe` version already takes care of this.